ostd/src/task/processor.rs

```diff
@@ -45,41 +47,47 @@ pub(super) fn switch_to_task(next_task: Arc<Task>) {
 
     let irq_guard = crate::trap::irq::disable_local();
 
-    let current_task_ptr = CURRENT_TASK_PTR.load();
-    let current_task_ctx_ptr = if !current_task_ptr.is_null() {
-        // SAFETY: The pointer is set by `switch_to_task` and is guaranteed to be
-        // built with `Arc::into_raw`. It will only be dropped as a previous task,
-        // so its reference will be valid until `after_switching_to`.
-        let current_task = unsafe { &*current_task_ptr };
-
-        // Until `after_switching_to`, the task's context is alive and can be exclusively used.
-        current_task.ctx.get()
-    } else {
-        // Until `after_switching_to`, IRQs are disabled and the context can be exclusively used.
-        BOOTSTRAP_CONTEXT.as_mut_ptr()
-    };
-
     before_switching_to(&next_task, &irq_guard);
 
     // `before_switching_to` guarantees that from now on, and while the next task is running on the
     // CPU, its context can be used exclusively.
     let next_task_ctx_ptr = next_task.ctx().get().cast_const();
 
+    let current_task_ptr = CURRENT_TASK_PTR.load();
     CURRENT_TASK_PTR.store(Arc::into_raw(next_task));
     debug_assert!(PREVIOUS_TASK_PTR.load().is_null());
     PREVIOUS_TASK_PTR.store(current_task_ptr);
 
     // We must disable IRQs when switching, see `after_switching_to`.
     core::mem::forget(irq_guard);
 
+    let current_task_ctx_ptr = if !current_task_ptr.is_null() {
+        // SAFETY: The pointer is set by `switch_to_task` and is guaranteed to be
+        // built with `Arc::into_raw`. It will only be dropped as a previous task,
+        // so its reference will be valid until `after_switching_to`.
+        let current_task = unsafe { &*current_task_ptr };
+
+        // Until `after_switching_to`, the task's context is alive and can be exclusively used.
+        current_task.ctx.get()
+    } else {
+        // `first_context_switch` performs the context switch to the first task on the current CPU.
+        // It will never return.
```

Comment:
Consider adding an `unreachable` macro behind `first_context_switch` to highlight this property.

Reply:
Note that the original approach sets the return type of `first_context_switch` to `!`. Therefore, adding an `unreachable!()` statement will prompt the compiler to issue warnings about unreachable code.

Since you suggested so, I have changed the return type of `first_context_switch` to `()` and added the explicit `unreachable!()` statement.